lib/rubygems.rb

```diff
@@ -963,8 +963,12 @@ def self.ruby_version
 
     if defined?(RUBY_PATCHLEVEL) && RUBY_PATCHLEVEL != -1 then
       version << ".#{RUBY_PATCHLEVEL}"
-    elsif defined?(RUBY_REVISION) then
-      version << ".dev.#{RUBY_REVISION}"
+    elsif defined?(RUBY_DESCRIPTION) then
+      if RUBY_ENGINE == "ruby" then
+        version << ".#{RUBY_DESCRIPTION.match(/\Aruby #{RUBY_VERSION}([^ ]+) /)[1]}"
```

Comment:
I think `Regexp.quote(RUBY_VERSION)` is better.
And I prefer `str[re, 1]` to `str.match(re)[1]`.

Reply:
Thanks for your review @znz. I addressed your feedback!